android/logto-client/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt

```diff
@@ -21,6 +25,96 @@ open class LogtoClient(
 
     fun isAuthenticated() = idToken != null
 
+    fun getAccessToken(callback: RetrieveCallback<AccessToken>) =
+        getAccessToken(null, null, callback)
+
+    fun getAccessToken(
+        resource: String?,
+        scope: List<String>?,
+        callback: RetrieveCallback<AccessToken>,
+    ) {
+        if (!isAuthenticated()) {
+            callback.onResult(LogtoException(LogtoException.Message.NOT_AUTHENTICATED), null)
+            return
+        }
+
+        resource?.let {
+            if (logtoConfig.resource?.contains(it) == false) {
+                callback.onResult(
+                    LogtoException(LogtoException.Message.RESOURCE_IS_NOT_GRANTED).apply { detail = it }, null
+                )
+                return
+            }
+        }
+
+        val finalScope = scope ?: logtoConfig.scope
+        if (!logtoConfig.scope.containsAll(finalScope)) {
+            callback.onResult(
+                LogtoException(LogtoException.Message.SCOPES_ARE_NOT_ALL_GRANTED).apply {
+                    detail = finalScope.toString()
+                },
+                null
+            )
+            return
+        }
+
+        // Retrieve access token from accessTokenMap
+        val accessTokenKey = buildAccessTokenKey(finalScope, resource)
+        val accessToken = accessTokenMap[accessTokenKey]
+        accessToken?.let {
```

Comment:
what if the accessToken is null?

Reply:
if the accessToken is null, then the code inside the `let` block will not be executed.